File: pset1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

// Prototypes
void print_row(int n, char s);

int main(void)
{
    // Prompt user for integer between 1 & 8
    int height = 0;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Draw pyramids of that height
    for (int i = 0; i < height; i++)
    {
        // Draw a row of right-aligned pyramid
        print_row(height - i - 2, ' '); // spaces
        print_row(i, '#');              // bricks

        // two spaces between pyramids
        printf("  ");

        // Draw a row if left-aligned pyramid
        print_row(i, '#');

        // New row
        printf("\n");
    }   
}

// Print a row of specific symbol
void print_row(int n, char s)
{
    for (int i = 0; i <= n; i++)
    {
        printf("%c", s);
    }
}

File: pset6/sentimental-mario-more/mario.py
def main():
    # Prompt user for a positive integer between 1 & 8
    height = 0

    while True:
        try:
            height = int(input("Height: "))
        except ValueError:
            height = 0
            pass

        if (height > 0) and (height <= 8):
            break
    
    # Draw pyramids
    draw_pyramids(height)


def draw_pyramids(h):
    for i in range(h):
        # Draw right-aligned pyramid
        print(" " * (h - i - 1), end="")  # Draw spaces
        print("#" * (i + 1), end="")     # Draw bricks

        # Two spaces between pyramids
        print("  ", end="")

        # Draw left-aligned pyramid with new row
        print("#" * (i + 1))


if __name__ == "__main__":
    main()